test/plugins/optional-auth/src/main/java/org/elasticsearch/test/plugin/optionalauth/OptionalAuthPlugin.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.test.plugin.optionalauth;
+
+import org.elasticsearch.action.support.ActionFilter;
+import org.elasticsearch.plugins.ActionPlugin;
+import org.elasticsearch.plugins.Plugin;
+
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+
+/**
+ * Plugin for testing authorization. See {@link OptionalAuthenticationDemandingActionFilter}.
+ */
+public class OptionalAuthPlugin extends Plugin implements ActionPlugin {
```

Comment:
I wonder if instead of using this plugin we should test the feature against a mock http server that requires authentication. Do we really need yet another full blown integration test for it?